.github/workflows/example-project.yml

```diff
@@ -35,13 +35,25 @@ jobs:
         run: |
           cargo install --path .
 
+      - name: Set MSVC/dumpbin path
```

Comment:
I would replace the entire block with

```
    - name: Set MSVC x86_64 linker path
      if: matrix.toolchain-suffix == '-msvc'
      run: |
        $LinkGlob = "VC\Tools\MSVC\*\bin\Hostx64\x64"
        $env:PATH = "$env:PATH;${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer"
        $LinkPath = vswhere -latest -products * -find "$LinkGlob" | Select-Object -Last 1
        echo "::add-path::$LinkPath"
```

Reply:
Sure, I can do that.

Just out of curiosity: why is that better?

Is that documented somewhere?

Reply:
The first line contains the path where `msvc` binaries are sought. I decided to assign the path to a new variable because that makes easier to change it eventually.
The second line adds the `vswhere` path to the `PATH` environment variable. 
The third and fourth lines are identical to the ones you have written.

Reply:
> I decided to assign the path to a new variable because that makes easier to change it eventually.

OK, that makes sense.
This way, the `vswhere` command can be written without a line break, which is nicer.

However, the variable name (containing "link") is confusing, isn't it?
AFAIU, this isn't really specific to "linking", but generally about a bunch of binaries (one of which I guess might be the linker?).

More specifically this is *not* a path used by the linker for looking up libraries, right?

> The second line adds the `vswhere` path to the `PATH` environment variable.

Is this necessary?
It seems to work fine without it.

> The third and fourth lines are identical to the ones you have written.

Except the variable name I used is more descriptive.

Reply:
> However, the variable name (containing "link") is confusing, isn't it?
> AFAIU, this isn't really specific to "linking", but generally about a bunch of binaries (one of which I guess might be the linker?).
> 
> More specifically this is _not_ a path used by the linker for looking up libraries, right?

Yep, exactly. Perhaps we may call it `VSBinariesGlob`, since it refers to the path containing `Visual Studio` binaries. What do you think? 

> Is this necessary?
> It seems to work fine without it.

Let's say it's an additional protection in case `GitHub` decides to remove the `vswhere` path from the `PATH` environment variable. Feel free to remove it if it works the same :)

> 
> Except the variable name I used is more descriptive.

Something line `VSBinariesPath`?

Reply:
I've added 1ab0e6fce226daf6890cd7a841362fb0d9c65636.

I chose the shorter name `$BinGlob`, because otherwise the line would have become overly long.

I've not added the unnecessary `PATH` setting; if it becomes necessary at some point in the future, we can add it then.

I've also changed to the new-style path-setting mechanism, see https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/.

Reply:
Perfect! I agree with your changes, thanks a lot! :)

Reply:
@mgeier 

It seems the CI fails to build on Windows, perhaps you should remove the `"` the `$BinGlob` variable on line 42

Reply:
It looks like setting the path with Powershell is ridiculously complicated, see https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files.

I've updated it in: 9ece0d32476e6823a4c7e5183b473d6cee4b5cce

> perhaps you should remove the `"` from the `$BinGlob` variable on line 42

I assumed this is necessary in case the path contains spaces?

But I'm not a PowerShell nor a Windows user at all, so I have no idea!

Reply:
You can add `shell: bash` if you fell more comfortable https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell

Reply:
Cool, I didn't know that!

I personally would indeed feel more comfortable with `bash`, but I think it would be good to use the "default" way of doing things in this example config.
I guess Windows users are more comfortable with PowerShell?
And it's the default, anyway.

What do you think?

Reply:
Yep, it's a valid reason, so let's use the `Powershell`